### PR TITLE
Fix a bug where outline does not update for inline elements

### LINKED ISSUE #433: Outline is not always updating the size or position
When changing margin or font size for example, outline doesn't update. When changing width/height - it does.

### PR DESCRIPTION
**Pull request by rpominov**
Fixes #433

## Before requesting a review

- [ ] if the PR is WIP - use draft mode
- [x] made a self-review
- [x] added inline comments where things may be not obvious (the "why", not "what")
- [ ] what kind of review is needed?
  - [x] conceptual
  - [ ] detailed
  - [ ] with testing

## Test cases

- [ ] step by step interaction description and what is expected to happen

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-designer/blob/main/apps/designer/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `designer/env-check.js` if mandatory

### CODE REVIEW

File: apps/designer/app/shared/dom-hooks/use-measure.ts [hunk 1]
```diff
@@ -19,11 +20,33 @@ export const useMeasure = <
...
+        return nextRect;
+      }
+      return currentRect;
+    });
   }, [element]);
 
+  // ResizeObserver does not work for inline elements,
+  // so if the element is inline, we measure it after any change in the instance tree.
+  //
+  // NOTE: We assume useMeasure is used only to measure elements on canvas.
```
Comment (kof) [inline apps/designer/app/shared/dom-hooks/use-measure.ts:46]: use-measure is in generic dom-hooks, so no, lets split up the generic part that can keep living here and the canvas specific part that would then move to canvas feature folder

File: packages/react-sdk/src/pubsub/create.ts [hunk 2]
```diff
@@ -66,5 +66,19 @@ export const createPubsub = <PublishMap>() => {
         };
       }, [type, onAction]);
     },
+
+    useSubscribeAll(
```
Comment (kof) [inline packages/react-sdk/src/pubsub/create.ts:70]: do we really need this function or could we just subscribe using asterix?

Comment (rpominov) [reply packages/react-sdk/src/pubsub/create.ts:70]: Just tricky with TS, I wouldn't spend time on trying to solve it

File: apps/designer/app/shared/dom-hooks/use-measure.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ export const useMeasure = <
   const [isInline, setIsInline] = useState(false);
 
   const handleChange = useCallback(() => {
-    if (element === null) return;
+    if (element === null || typeof window === "undefined") return;
```
Comment (kof) [inline apps/designer/app/shared/dom-hooks/use-measure.ts:26]: how can element be not null on the server? I mean there is no way it is being set serverside lol

Comment (rpominov) [reply apps/designer/app/shared/dom-hooks/use-measure.ts]: Yeah, was just trying anything when I was getting the confusing error

File: apps/designer/app/canvas/shared/instance.ts [hunk 4]
```diff
@@ -36,7 +36,7 @@ import {
   useRootInstance,
   useTextEditingInstanceId,
 } from "~/shared/nano-states";
-import { useMeasure } from "~/shared/dom-hooks";
+import { useMeasure } from "~/canvas/shared/use-measure";
```
Comment (kof) [inline apps/designer/app/canvas/shared/instance.ts:39]: maybe use-measure-instance just to be clear its about instance specifically

File: apps/designer/app/canvas/shared/use-measure.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+import { useCallback, useEffect, useMemo } from "react";
+import { useMeasure as useMeasureBase, useRectState } from "~/shared/dom-hooks";
+import { type PubsubMap, useSubscribeAll } from "~/shared/pubsub";
+
+const useTreeChange = (onChange: () => void, enabled: boolean) => {
```
[the remark this answers is not in the document]
Comment (kof) [reply apps/designer/app/canvas/shared/use-measure.ts:5]: useSubscribeTreeChange

**Review by kof — APPROVED** [review]
looking good, a few naming improvements

**Pull request merged**